rig-core/src/providers/gemini/transcription.rs

```diff
@@ -0,0 +1,144 @@
+use std::path::Path;
+
+use base64::{prelude::BASE64_STANDARD, Engine};
+use mime_guess;
+use serde_json::{Map, Value};
+
+use crate::{
+    providers::gemini::completion::gemini_api_types::{
+        Blob, Content, GenerateContentRequest, GenerationConfig, Part, Role,
+    },
+    transcription::{self, TranscriptionError},
+    OneOrMany,
+};
+
+use super::{completion::gemini_api_types::GenerateContentResponse, Client};
+
+pub use super::completion::{
+    GEMINI_1_5_FLASH, GEMINI_1_5_PRO, GEMINI_1_5_PRO_8B, GEMINI_2_0_FLASH,
+};
+
+#[derive(Clone)]
+pub struct TranscriptionModel {
+    client: Client,
+    /// Name of the model (e.g.: gemini-1.5-flash)
+    pub model: String,
+}
+
+impl TranscriptionModel {
+    pub fn new(client: Client, model: &str) -> Self {
+        Self {
+            client,
+            model: model.to_string(),
+        }
+    }
+}
+
+impl transcription::TranscriptionModel for TranscriptionModel {
+    type Response = GenerateContentResponse;
+
+    #[cfg_attr(feature = "worker", worker::send)]
+    async fn transcription(
+        &self,
+        request: transcription::TranscriptionRequest,
+    ) -> Result<
+        transcription::TranscriptionResponse<Self::Response>,
+        transcription::TranscriptionError,
+    > {
+        // Handle Gemini specific parameters
+        let additional_params = request
+            .additional_params
+            .unwrap_or_else(|| Value::Object(Map::new()));
+        let mut generation_config = serde_json::from_value::<GenerationConfig>(additional_params)?;
+
+        // Set temperature from completion_request or additional_params
+        if let Some(temp) = request.temperature {
+            generation_config.temperature = Some(temp);
+        }
+
+        let system_instruction = Some(Content {
+            parts: OneOrMany::one(
+                "Translate the provided audio exactly. Do not add additional information.".into(),
+            ),
+            role: Some(Role::Model),
+        });
+
+        let mime_type =
+            if let Some(mime) = mime_guess::from_path(Path::new(&request.filename)).first() {
```

Comment:
`mime_guess` looks neat, it might be worthwhile for us to switch to using `mime` throughout the entire codebase in general instead of my custom set in the `completion/messages.rs`